codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/protocol/ServerProtocolTestGenerator.kt

```diff
@@ -372,20 +481,45 @@ class ServerProtocolTestGenerator(
         }
     }
 
-    private fun checkParams(httpRequestTestCase: HttpRequestTestCase, rustWriter: RustWriter) {
-        rustWriter.writeInline("let expected = ")
-        instantiator.render(rustWriter, inputShape, httpRequestTestCase.params)
-        rustWriter.write(";")
+    private fun checkRequest(operationShape: OperationShape, operationSymbol: Symbol, httpRequestTestCase: HttpRequestTestCase, rustWriter: RustWriter) {
+        val inputShape = operationShape.inputShape(coreCodegenContext.model)
+        val outputShape = operationShape.outputShape(coreCodegenContext.model)
 
         val operationName = "${operationSymbol.name}${ServerHttpBoundProtocolGenerator.OPERATION_INPUT_WRAPPER_SUFFIX}"
-        rustWriter.rustTemplate(
+        val (inputT, outputT) = operationInputOutputTypes[operationShape]!!
+        val helperNamespace = "crate::operation::protocol_test_helper"
+        rustWriter.withBlock(
             """
-            let mut http_request = #{SmithyHttpServer}::request::RequestParts::new(http_request);
-            let parsed = super::$operationName::from_request(&mut http_request).await.expect("failed to parse request").0;
+            $helperNamespace::validate_request(
+                http_request,
+                &|builder| {
+                    builder.${operationShape.toName()}((|input| Box::pin(async move {
             """,
-            *codegenScope,
-        )
 
+            "})) as $helperNamespace::F<$inputT, $outputT>)}).await",
+
+        ) {
+            // Construct expected request
+            rustWriter.writeInline("let expected = ")
+            instantiator.render(rustWriter, inputShape, httpRequestTestCase.params)
+            rustWriter.write(";")
```

Comment:
```suggestion
            rustWriter.withBlock("let expected = ", ";") {
                instantiator.render(this, inputShape, httpRequestTestCase.params)
            }
```

This pattern is handy to not have to keep track of the trailing `;`.